cmake/opentelemetry-proto.cmake

```diff
@@ -108,113 +102,120 @@ set(METRICS_SERVICE_PB_H_FILE
     "${GENERATED_PROTOBUF_PATH}/opentelemetry/proto/collector/metrics/v1/metrics_service.pb.h"
 )
 if(WITH_OTLP_GRPC)
-set(METRICS_SERVICE_GRPC_PB_CPP_FILE
-    "${GENERATED_PROTOBUF_PATH}/opentelemetry/proto/collector/metrics/v1/metrics_service.grpc.pb.cc"
-)
-set(METRICS_SERVICE_GRPC_PB_H_FILE
-    "${GENERATED_PROTOBUF_PATH}/opentelemetry/proto/collector/metrics/v1/metrics_service.grpc.pb.h"
-)
+  set(METRICS_SERVICE_GRPC_PB_CPP_FILE
+      "${GENERATED_PROTOBUF_PATH}/opentelemetry/proto/collector/metrics/v1/metrics_service.grpc.pb.cc"
+  )
+  set(METRICS_SERVICE_GRPC_PB_H_FILE
+      "${GENERATED_PROTOBUF_PATH}/opentelemetry/proto/collector/metrics/v1/metrics_service.grpc.pb.h"
+  )
 endif()
 
 foreach(IMPORT_DIR ${PROTOBUF_IMPORT_DIRS})
   list(APPEND PROTOBUF_INCLUDE_FLAGS "-I${IMPORT_DIR}")
 endforeach()
 
 if(WITH_OTLP_GRPC)
-if(CMAKE_CROSSCOMPILING)
+  if(CMAKE_CROSSCOMPILING)
     find_program(gRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
-else()
-    set(gRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:gRPC::grpc_cpp_plugin>)
-endif()
+  else()
+    if(TARGET gRPC::grpc_cpp_plugin)
+      project_build_tools_get_imported_location(gRPC_CPP_PLUGIN_EXECUTABLE
```

Comment:
why do we need this - Wasn't `set(gRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:gRPC::grpc_cpp_plugin>)` sufficient here?

Reply:
I want to print the real path of grpc_cpp_plugin here, just like protobuf executable.

> BTW: In the build system of our projects, it's also useful to use the same grpc_cpp_plugin to build other packages.Thers is no simular problem here in opentelemetry-cpp because it use vcpkg to install dependencies.